### PR TITLE
feat: add jwt and backend login

### PR DESCRIPTION
**Pull request by MorveN11**
# Pull Request

## Description

This pull request is in charge of making the jwt feature with roles, differentiating the actions that could only have an admin and a common user, at the same time we made a refactor of file names and folders to maintain greater consistency in the code, we added the jwt and authentication to all paths, and finally we made an update of the dependencies.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: projects/api/prisma/seed.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ async function seed() {
 
 seed()
   .catch((e) => {
+    // eslint-disable-next-line no-console
     console.error('Error seeding database', e);
```
Comment (rencinasf) [inline projects/api/prisma/seed.ts:44]: console.error? remove this line and add logging.Error

File: projects/api/src/auth/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import CreateUserDto from '@/user/dto/create-user.dto';
+
+import AuthService from './auth.service';
+import LoginDto from './dto/login.dto';
+import Auth from './entities/auth.entity';
+
+@ApiTags('Auth')
+@Controller('auth')
+class AuthController {
```
Comment (rencinasf) [inline projects/api/src/auth/auth.controller.ts:12]: Auth? Please fill the name at all, something like this: AuthenticationController

Comment (rencinasf) [reply projects/api/src/auth/auth.controller.ts:12]: hmm I was not able to see... strategies. those should be used for authentication...

File: projects/api/prisma/schema.prisma [hunk 3]
```diff
@@ -19,33 +30,174 @@ model Country {
...
+  updatedAt     DateTime        @default(now()) @updatedAt @map(name: "updated_at") @db.Timestamptz()
+  penalty       Penalty?
+  wishList      WishList?
+  purchases     Purchase[]
+  usersProducts UsersProducts[]
+
+  @@map(name: "users")
+}
+
+model Penalty {
```
Comment (rencinasf) [inline projects/api/prisma/schema.prisma:84]: What does it mean?

File: projects/api/prisma/schema.prisma [hunk 4]
```diff
@@ -19,33 +30,174 @@ model Country {
...
+  id        String   @id @default(uuid()) @db.Uuid()
+  amount    Int      @default(0)
+  createdAt DateTime @default(now()) @map(name: "created_at") @db.Timestamptz()
+  updatedAt DateTime @default(now()) @updatedAt @map(name: "updated_at") @db.Timestamptz()
+  userId    String   @unique @map(name: "user_id") @db.Uuid()
+  user      User     @relation(fields: [userId], references: [id], onDelete: Restrict, onUpdate: Cascade)
```
Comment (rencinasf) [inline projects/api/prisma/schema.prisma:90]: If you have the userId, why the user is needed at all?

File: projects/api/prisma/schema.prisma [hunk 5]
```diff
@@ -19,33 +30,174 @@ model Country {
...
+
+  @@map(name: "penalties")
+}
+
+model WishList {
+  id                String              @id @default(uuid()) @db.Uuid()
+  title             String              @default("My WishList")
+  description       String              @default("")
+  createdAt         DateTime            @default(now()) @map(name: "created_at") @db.Timestamptz()
+  updatedAt         DateTime            @default(now()) @updatedAt @map(name: "updated_at") @db.Timestamptz()
```
Comment (rencinasf) [inline projects/api/prisma/schema.prisma:100]: Why does the updatedAt have a value by default if it is not updated at the first time?

File: projects/api/prisma/schema.prisma [hunk 6]
```diff
@@ -19,33 +30,174 @@ model Country {
...
+
+  @@map(name: "wish_lists")
+}
+
+model Purchase {
+  id                String              @id @default(uuid()) @db.Uuid()
+  totalCost         Float               @map(name: "total_cost")
+  deliveyDay        DateTime            @map(name: "delivery_day")
+  deadline          DateTime
+  status            PurchaseStatus
```
Comment (rencinasf) [inline projects/api/prisma/schema.prisma:113]: What are the purchasing states you are considering?

File: projects/api/prisma/schema.prisma [hunk 7]
```diff
@@ -19,33 +30,174 @@ model Country {
...
+}
+
+model Product {
+  id                String              @id @default(uuid()) @db.Uuid()
+  name              String              @unique @db.Citext
+  description       String              @default("")
+  stock             Int
+  price             Float
+  isAvailable       Boolean             @default(true) @map(name: "is_available")
+  createdAt         DateTime            @default(now()) @map(name: "created_at") @db.Timestamptz()
```
Comment (rencinasf) [inline projects/api/prisma/schema.prisma:130]: hmmm why are you using timestamptz and not timestamp?

File: projects/api/src/auth/auth.controller.ts (hunk 2 above)
Comment (MorveN11) [reply projects/api/src/auth/auth.controller.ts:14]: all the strategies are in a directory named strategy, and its Auth is only the name that appears in the swagger not more else

File: projects/api/prisma/schema.prisma (hunk 4 above)
Comment (MorveN11) [reply projects/api/prisma/schema.prisma:90]: the user field is only a field of prisma to declare the relation between the tables, is not a field in out table, maybe for more info can review the init.sql in the scripts directory

File: projects/api/prisma/schema.prisma (hunk 5 above)
Comment (MorveN11) [reply projects/api/prisma/schema.prisma:100]: we will fix that in a moment

**Review by Severaldog — APPROVED** [review]

**Pull request merged**